docker-new/README.md

```diff
@@ -0,0 +1,238 @@
+        --name ${USER}/ccf_container pdo_ccf_base
+```
+
+The `start_development.sh` script can be used in the `ccf_container`
+in the same way it is used in the `services_container` above.
+
+Note: be sure to run `make clean` in the original (out of container) build
+tree.
+
+## Pattern: Service Deployment ##
+
+The containers can be used to deploy network services from the
+`pdo_ccf` and `pdo_services` images. Set the environment variables
+`PDO_HOSTNAME` to the interface where the service will listen and
+`PDO_LEDGER_URL` as the endpoint for CCF.
+
+** NOTE: We need a better way to process registrations for SGX HW mode. In
+theory, the best way to do this may be to create a canonical base
+services image; populate an instance of it with CCF private keys, run
+the registration. That way the canonical base service image would have
+a standard version of the enclave library that would not have to deal
+with reproducible builds.**
+
+### CCF Deployment ###
+
+```bash
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}/ccf_container pdo_ccf
+```
+
+This will configure the CCF service using the default configuration
+based on `PDO_HOSTNAME` and `PDO_LEDGER_URL` for defining the
+endpoint.
+
+You may also run the CCF with a pre-built configuration. Create the
+directores `$(SCRIPT_DIR)/xfer/ccf/etc` and
+`$(SCRIPT_DIR)/xfer/ccf/keys`. Copy the CCF configuration files
+(`cchost.toml` and `constitution.js`) into the `etc` directory.
+
+```bash
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}/ccf_container pdo_ccf --mode copy
+```
+
+The CCF container will create the `networkcert.pem` key file and
+place it in the `$(SCRIPTDIR)/xfer/ccf/keys`.
+
+** NOTE: We do not support starting a CCF instance to join an
+existing CCF network. This is future work. **
+
+### PDO Services Deployment ###
+
+Before starting the services container, be sure to copy the CCF ledger
+keys into `${PDO_SOURCE_ROOT}/docker-new/xfer/ccf/keys`. Those keys
+should be available from that directory on the host where the CCF
+ledger is running.
+
+```bash
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}/services_container pdo_services
+```
+
+This will configure and create the standard set of five `eservices`,
+five `pservices` and five `sservices`. It will take `PDO_HOSTNAME` and
+`PDO_LEDGER_URL` that existed when the `pdo_services` image was built.
+These can be overridden by adding parameters to the `docker-run`
+command:
+
+```bash
+    docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host \
+        --name ${USER}/services_container pdo_services --ledger <URL> --interface <HOST>
+```
+
+You may also run the services with a pre-built configuration. Create
+the directores `$(SCRIPT_DIR)/xfer/services/etc` and
+`$(SCRIPT_DIR)/xfer/services/keys`. Copy the service configuration
```

Comment:
A few questions here:
1. How do I build a configuration? I suppose you mean the toml files for the services that you want to start? Or is it site.psh?
2. It would be good if the container could share its "configuration fiels in use" with the xfrer/services/etc folder if these are generated by the container. The admin could save this locally, modify and use next time with copy mode.
3. It would be good to add documentation on how to modify the above command to start a service with a single service.

Reply:
Regarding 1... that is through pdo-services-config, pdo-ccf-config, and pdo-client-config (which is why those tools exist). 

Regarding 2... maybe... its pretty easy to mount the etc directory when you start the container to see what goes into it so i don't think anything needs to change. there is a basic assumption here about docker flexibility. it is not our responsibility to document every possible way to use docker. that's why the documentation talks about patterns. what you ask for is easily done with docker without any changes to the code.

Regarding 3... again, its a pattern we can add to the documentation (i'm adding a "how to set up for developing contracts" pattern to the documentation that will be pushed shortly). however, this is not supposed to be an all inclusive documentation of every possible thing you could do.